app/controllers/posts_controller.rb

```diff
@@ -11,5 +11,14 @@ def show
     @user = User.find_by(id: params['user_id']).posts.find_by(id: params['id'])
     @comment = Comment.where(post_id: @posts)
     @comment = Comment.where(author_id: @user.id)
+    @posts = User.find_by(id: params['user_id']).posts.find_by(id: params['id'])
+  end
+
+  def create
+    @values = params[:post]
+    @post = Post.new(title: @values['title'], text: @values['text'], author: User.find_by(id: params['user_id']))
+    return unless @post.save
+
+    redirect_to "/users/#{params['user_id']}"
   end
```

Comment:
- [ ] Why creating  unneccesary variable "@values", you can directly use params[:post], kIndly use directly

```ruby
def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end

    private 

    def post_params
      params.require(:post).permit(:title, :text)
    end
```

- [ ] OPTIONAL -> Use of strong parameter will help you to increase security level. Kindly learn about them & use it here.